rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,562 @@
+---
+feature: doc-comment-standard
+start-date: 2023-03-27
+author: hsjobeki
+co-authors: --
+shepherd-team: @DavHau; @sternenseemann; @asymmetric
+shepherd-leader: @lassulus
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Propose a standard format specification for Doc-comments.
+
+This RFC includes two concerns that define a doc-comment:
+
+- Outer format rules to allow distinction between regular comments and doc-comments
+- Inner format rules that describe the required format of the content.
+
+However, both concerns relate closely to each other; It makes sense and reduces bureaucracy to address that in a single RFC.
+
+# Motivation
+[motivation]: #motivation
+
+The following are the envisioned goals.
+
+- Be able to generate documentation from code for any nix expression.
+- Make doc comments easy to write and read
+- Standardize a format for doc comments that further RFCs can extend
+  - E.g. extend with code examples that can be converted to tests (aka doctests).
+
+This RFC is a significant change to existing documentation conventions.
+It allows distinguishing between regular and doc comments. Having distinction is essential because arbitrary code comments should not end up in generated documentation.
+
+> Note: Generating static documentation is a controversial topic in nixpkgs.[^1][^2][^3][^4] We found that it is very hard to generate accurate documentation statically. A correct solution would involve the evaluation of expressions in some way.
+
+[^1]: [Unmerged PR: nix repl doc-comments](https://github.com/NixOS/nix/pull/1652), 2017
+[^2]: [Unmerged PR: doc-strings for functors](https://github.com/NixOS/nix/pull/5527), 2021
+[^3]: [discourse: users asking how to obtain documentation](https://discourse.nixos.org/t/how-to-generate-documentation-from-arbitrary-nix-code/22292), 2022
+[^4]: [discourse: aksing for consensus on documentation](https://discourse.nixos.org/t/any-consensus-on-documentation-generation-tool-for-nixpkgs-manual/15550), 2021
+
```

Comment:
I would add here:

```suggestion

# Non-goals

- Discuss in which tool doc-comments are parsed and rendered. This could be an external tool, or Nix, or something else entirely, but that's out of scope for this RFC.
- Providing a migration path for existing comments. This is expected to require some amount of manual work. See "Future work" section.
```